pandas/core/indexes/base.py

```diff
@@ -230,7 +230,7 @@ def _outer_indexer(self, left, right):
         return libjoin.outer_join_indexer(left, right)
 
     _typ = "index"
-    _data = None
+    _data: Union[ExtensionArray, np.ndarray]
```

Comment:
you can now remove the assert and comment added in #29475 on L657?

Reply:
yes, updated

Reply:
updated+green